packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -1007,8 +1007,9 @@ class _AdbLogReader extends DeviceLogReader {
   String get name => device.name;
 
   void _start() {
-    // Start the adb logcat process and filter logs by the "flutter" tag.
-    final List<String> args = <String>['shell', '-x', 'logcat', '-v', 'time', '-s', 'flutter'];
+    final String lastTimestamp = device.lastLogcatTimestamp;
+    // Start the adb logcat process and filter the most recent logs since `lastTimestamp`.
+    final List<String> args = <String>['logcat', '-v', 'time', '-T', lastTimestamp];
     processUtils.start(device.adbCommandForDevice(args)).then<void>((Process process) {
```

Comment:
this looks a lot simpler than the code used to look; why was the old code more complicated?

Reply:
I'm not sure. I was trying to see in which version the `-T` flag was introduced.